polkadot/runtime/common/src/slots/mod.rs

```diff
@@ -221,6 +241,52 @@ pub mod pallet {
 			};
 			Ok(())
 		}
+
+		/// Try to refund the lease deposit before the actual end of the lease.
+		///
+		/// This is only allowed if lease is ending within [`Config::EarliestRefundPeriod`] and the
+		/// parachain has no new lease periods coming up. This is useful for parachains who want to
+		/// get access to their funds they used in the last lease and rebid using same for the
+		/// next lease.
+		///
+		/// Can only be called by the parachain manager.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::early_lease_refund())]
+		pub fn early_lease_refund(origin: OriginFor<T>, para: ParaId) -> DispatchResult {
+			// ensure caller is ForceOrigin or the manager of the parachain.
+			ensure_signed(origin.clone())
+				.map_err(|e| e.into())
+				.and_then(|who| -> DispatchResult {
+					ensure!(Some(who) == T::Registrar::manager_of(para), Error::<T>::NoPermission);
+					Ok(())
```

Comment:
This allows the manager of the Parachain on the Relay Chain to call the function, right? If so, should we instead make it the parachain itself? They could make a call from the parachain via UMP, as it happens today with `swap`.

Reply:
Yes this will allow parachain manager to call it (or ForceOrigin which can be root or lease admin). Do you want to allow  all three: root, para and para_manager?

Reply:
I'm ok either way! Usually the way I've seen this in the past is that the para_manager is set to do more super-admin like stuff, when a parachain is unlocked (like changing wasm/head), however on normal operations the goal is to use the parachain itself to send the UMP. But happy to learn from other opinions here too.